src/client/Client.h

```diff
@@ -534,6 +534,8 @@ class Client : public Dispatcher, public md_config_obs_t {
    */
   void _handle_full_flag(int64_t pool);
 
+  void close_sessions();
```

Comment:
Oh, can you switch this to be _close_sessions()? That's usually how we indicate it needs to be called with locks held in the Client.